numba/runtime/nrt.c

```diff
@@ -213,6 +223,16 @@ MemInfo* NRT_MemInfo_new(void *data, size_t size, dtor_function dtor,
     return mi;
 }
 
+size_t NRT_MemInfo_refcount(MemInfo *mi) {
+    /* Should never returns 0 for a valid MemInfo */
+    if (mi && mi->payload.data)
+        return mi->payload.refct;
+    else{
+        assert (0 && "Unreachable: MemInfo is invalid");
```

Comment:
Since it's a debugging facility, how about turning it into an exception?
(you could make the function return a Py_ssize_t and return -1 here)